pkg/scheduler/objects/application.go

```diff
@@ -886,6 +886,40 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, nodeIterator fu
 					zap.String("required node", requiredNode))
 				return nil
 			}
+
+			// Are there any non daemon set reservations on specific required node?
+			// Cancel those reservations to run daemon set pods
+			reservations := node.getReservations()
+			if len(reservations) > 0 {
+				unreserve := true
+				for _, res := range node.getReservations() {
+					// skip the node
```

Comment:
nit: can be fixed on commit, explain why we skip the node. Even I had to think twice why we would skip.